mtypes/iso8601.go

```diff
@@ -0,0 +1,65 @@
+package mtypes
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/mailgun/errors"
+)
+
+// ISO8601Time Mailgun uses ISO8601 format for timestamps in the API key API endpoints ('2011/10/13T18:02:00'), but
+// by default Go's JSON package uses another format when decoding/encoding timestamps. This goes against the documentation
+// at https://documentation.mailgun.com/docs/mailgun/api-reference#date-format but here we are.
+type ISO8601Time time.Time
```

Comment:
Could you make:
```suggestion
type ISO8601Time struct {
    time.Time
}
```
So it inherits all `time.Time` methods, like `Unix() int64` and `IsZero() bool`.

Reply:
> I'd recommend updating RFC2822Time as well at some point to conform with the above.

Yeah. Forgot to do it in [v5](https://github.com/mailgun/mailgun-go/pull/367). Need to add some TODO for v6.